Add Devant concepts docs (#21)
wso2/docs-devant: en/docs/devant-concepts/data-planes.md

# Data Planes

> Devant's architecture comprises two key components: the control plane and the data plane. The control plane handles essential tasks such as administering organizations, users, and projects. In addition, it also governs the entire journey of integration development, from the initial stages of creation, progressing to deployment, including measures to enforce governance and the provision for observability. The Devant control plane is a SaaS that manages all cloud data planes and private data planes. It caters to diverse user personas, including CIOs, architects, and developers, as well as DevOps, and platform engineers.

Reviewer: ```suggestion
Devant's architecture comprises two key components: the control plane and the data plane. The control plane handles essential tasks such as administering organizations, users, and projects. In addition, it also governs the entire journey of integration development, from the initial stages of creation and progress to deployment, including measures to enforce governance and the provision for observability. The Devant control plane is a SaaS that manages all cloud and private data planes. It caters to diverse user personas, including CIOs, architects, and developers, as well as DevOps, and platform engineers.
```

> The data plane is the environment where user integrations are deployed based on configurations set in the control plane. The integrations can be built in [Ballerina](https://ballerina.io/) and [Wso2 Micro Integrator](https://wso2.com/integrator/micro-integrator/) which are optimized for building integrations or using the Devant Cloud Editor. Importantly, all traffic related to the runtime of user integrations are restricted to the Devant data plane, ensuring strict containment of user data within its boundaries.

Reviewer: ```suggestion
The data plane is the environment where user integrations are deployed based on configurations set in the control plane. The integrations can be built in [Ballerina](https://ballerina.io/) and [Wso2 Micro Integrator](https://wso2.com/integrator/micro-integrator/) which are optimized for building integrations or using the Devant Cloud Editor. Importantly, all traffic related to the runtime of user integrations is restricted to the Devant data plane, ensuring strict containment of user data within its boundaries.
```

<td>Devant control plane (CP) (mizzen server)</td>
<td>Outbound</td>
<td>WSS</td>
</tr>
<tr>
<td>Kubernetes API server</td>
<td>Outbound (cluster internal)</td>
<td>HTTPS, WS</td>
</tr>
<tr>
<td rowspan=2>APIM/local adaptor</td>
<td>Global adaptor</td>
<td>Outbound</td>
<td>HTTPS</td>
</tr>
<tr>
<td>Azure Service Bus (CP)</td>
<td>Outbound</td>
<td>AMQP</td>
</tr>
<tr>
<td >APIM/Enforcer</td>
<td>Event hub (CP)</td>
<td>Outbound</td>
<td>AMQP</td>
</tr>
<tr>
<td>Devant secret resolver</td>
<td>Cloud secret store</td>
<td>Outbound (VPC internal)</td>
<td>HTTPS</td>
</tr>
<tr>
<td rowspan=2>Container registry</td>
<td>Container registry (public)</td>
<td>Inbound</td>
<td>HTTPS</td>
</tr>
<tr>
<td>Container registry</td>
<td>Outbound (VPC internal)</td>
<td>HTTPS</td>
</tr>
<tr>
<td rowspan=2>Certificate manager</td>
<td>Azure DNS service</td>
<td>Outbound</td>
<td>HTTPS</td>
</tr>
<tr>
<td>LetsEncrypt</td>
<td>Outbound</td>
<td>HTTPS</td>
</tr>
<tr>
<td>Flux source controller</td>
<td>GitHub</td>
<td>Outbound</td>
<td>HTTPS</td>
</tr>
<tr>
<td>Flux Helm controller</td>
<td>Devant container registry</td>
<td>Outbound</td>
<td>HTTPS</td>
</tr>
</tbody>
</table> 

> All communication between the control plane and the private data plane is secured using TLS.

Reviewer: ```suggestion
All communication between the control and private data planes is secured using TLS.
```

### Observability architecture

The following diagram depicts the architecture overview of Devant's in-data-plane log and observability in Azure PDP:

[//]: # (Tooo: Modify this diagram to match Devant)
[//]: # (![Observability architecture]&#40;../assets/img/devant-concepts/observability-architecture.png&#41;)

> The private data plane observability architecture is centered around a strong commitment to data privacy and compliance. This is achieved through a strategic decision to retain logs and observability data within the data planes itself. Key aspects of this architecture include:

Reviewer: ```suggestion
The private data plane observability architecture centers around a strong commitment to data privacy and compliance. A strategic decision to retain logs and observability data within the data planes achieves this. Key aspects of this architecture include:
```

- **Data storage at source**: Logs and observability data are stored within the data plane itself, enhancing security, simplifying access, and ensuring compliance.
- **Direct browser-to-data-plane interaction**: The Devant Console in the user's browser directly interacts with APIs in the data plane, reducing potential data routing complexities and ensuring a more secure, direct flow of information.
- **Reduced data exposure points**: Fetching data directly from the data plane's APIs minimizes the number of data transfer points, effectively decreasing the chances of data exposure or interception.
- **Compliance with regulatory standards**: The architecture supports data locality, aligning with global regulatory standards like GDPR and CCPA by keeping data in its original environment.
- **Improved performance and real-time insights**: Direct interaction between the browser and data plane results in faster data retrieval, providing users with immediate insights.
- **User transparency and control**: Users have a clear view of their data's location and access methods, alongside granular control over data access.

### Security

> The Devant private data plane ensures extensive, production-grade security, ranging from infrastructure and architecture to zero-trust network security. All incoming traffic is protected by a firewall and must undergo authentication and authorization via the API Gateway. It also provides end-to-end network traffic encryption using Cilium transparent encryption, ensuring efficient data path encryption.

Reviewer: ```suggestion
The Devant private data plane ensures extensive, production-grade security, ranging from infrastructure and architecture to zero-trust network security. A firewall protects all incoming traffic and must undergo authentication and authorization via the API Gateway. It also provides end-to-end network traffic encryption using Cilium transparent encryption, ensuring efficient data path encryption.
```
